plugin/storage/es/mappings/mapping.go

```diff
@@ -63,24 +73,60 @@ func (mb MappingBuilder) getMappingTemplateOptions(mapping string) templateParam
 	return mappingOpts
 }
 
+func (mt MappingType) String() string {
+	switch mt {
+	case SpanMapping:
+		return "jaeger-span"
+	case ServiceMapping:
+		return "jaeger-service"
+	case DependenciesMapping:
+		return "jaeger-dependencies"
+	case SamplingMapping:
+		return "jaeger-sampling"
+	default:
+		return "unknown"
+	}
+}
+
+// MappingTypeFromString converts a string to a MappingType
+func MappingTypeFromString(val string) (MappingType, error) {
+	switch val {
+	case "jaeger-span":
+		return SpanMapping, nil
+	case "jaeger-service":
+		return ServiceMapping, nil
+	case "jaeger-dependencies":
+		return DependenciesMapping, nil
+	case "jaeger-sampling":
+		return SamplingMapping, nil
+	default:
+		return -1, fmt.Errorf("invalid mapping type: %s", val)
+	}
+}
+
 // GetMapping returns the rendered mapping based on elasticsearch version
-func (mb *MappingBuilder) GetMapping(mapping string) (string, error) {
-	templateOpts := mb.getMappingTemplateOptions(mapping)
-	if mb.EsVersion == 8 {
-		return mb.renderMapping(mapping+"-8.json", templateOpts)
-	} else if mb.EsVersion == 7 {
-		return mb.renderMapping(mapping+"-7.json", templateOpts)
+func (mb *MappingBuilder) GetMapping(mappingType MappingType) (string, error) {
+	templateOpts := mb.getMappingTemplateOptions(mappingType)
+	var version int
```

Comment:
What is the point of this variable?

Reply:
Tried assigning the ES version number to it via this variable instead of doing `-6`, `-7`, `-8` explicitly